workshops/chrome-extension/background.js

```diff
@@ -0,0 +1,74 @@
+let timer;
+
+chrome.browserAction.setBadgeBackgroundColor({ color: '#777' });
+
+// retrieve and format timer text or play audio if time is up
+function getTimeLeft(endTime) {
+  const distance = new Date(endTime) - new Date();
+  if (distance <= 0) {
+    timerEnded();
+    return '';
+  }
+  const hours = Math.floor(distance / (1000 * 3600));
+  const minutes = Math.floor((distance % (1000 * 3600)) / (1000 * 60));
+  const seconds = Math.floor((distance % (1000 * 60)) / 1000);
+  let message = '';
+  // if multiple hours remain, format as "hh hrs" to prevent awkward UI
+  if (hours > 0) {
+    return `${hours} hrs`;
+  }
+  if (minutes >= 0) message += minutes === 0 ? `00:` : minutes < 10 ? `0${minutes}:` : `${minutes}:`;
+  if (seconds >= 0) message += seconds === 0 ? `00` : seconds < 10 ? `0${seconds}` : `${seconds}`;
+  return message;
+}
+
+function timerEnded() {
+  const myAudio = new Audio(chrome.runtime.getURL('old_town_road.mp3'));
+  myAudio.play();
+  clearInterval(timer);
+  chrome.tabs.executeScript({
+    code: `alert('Your timer has ended!');`
+  });
+}
+
+const getStorageData = key =>
+  new Promise((resolve, reject) =>
+    chrome.storage.sync.get([key], function(result) {
+      chrome.runtime.lastError ? reject(Error(chrome.runtime.lastError.message)) : resolve(result[key]);
+    })
+  );
+
+// update badge timer text
+async function setBadgeTimer() {
+  let timerStartTime;
+  let timerEndTime;
+  await getStorageData('startTime').then(result => (timerStartTime = result));
+  await getStorageData('endTime').then(result => (timerEndTime = result));
+  // start timer if start time exists
+  if (timerStartTime && timerStartTime !== 0) {
```

Comment:
Technically you don't need `&& timerStartTime !== 0` because 0 is falsy in Javascript